docs/react-v9/contributing/rfcs/react-components/convergence/extended-design-tokens.md

```diff
@@ -0,0 +1,455 @@
+| Property  | A visual attribute of the Component or Part. Often maps to a CSS property, but not always (e.g. stroke)                                        | ForegroundColor<br>BackgroundColor<br>IconAfter<br>StrokeWidth |
+| Modifier  | Context that changes the value of a property based on state or mode. Usually maps to an interactive state or mode.                             | Rest<br>Hover<br>Pressed<br>Disabled<br>Selected-\*            |
+
+_Variants_
+
+The term “variants” is often used to describe alternative forms or styles of components. There are typically 3 types of variants representing different axes of changes:
+
+1.  **Style variant**: Component has a different look and feel, but same behaviors and visual structure as the base component.
+
+    a. In code, v9 typically uses appearance prop
+
+    b. In Figma, the Style property controls these
+
+    c. Example: Button has Secondary, Primary, Subtle, Transparent style variants
+
+2.  **Layout variant**: Component has a different visual structure from the base component, but typically retains the same behaviors and data. Different Parts may be toggled on/off or have different visual treatments due to their composition.
+
+    a. In code, this might be controlled through optional component props e.g. Button icon and iconPosition
+
+    b. In Figma,
+
+    c. Example: Buttons can have just a text label, just an icon, or an icon and Button
+
+3.  **Functional variant**: Component exposes different interactions or data than the base component.
+
+    a. In code, these might be separate components with different APIs
+
+## Expose expanded tokens in Theme
+
+Each component that exposes tokens would add a Component.tokens.ts file to expose its tokens API:
+
+```javascript
+// Button.tokens.ts
+
+import { tokens } from '@fluentui/react-theme';
+
+// Semantic button, shared between all Button-family components
+
+export const buttonGroupTokens = {
+
+buttonBorderRadius: \`var(--buttonBorderRadius, ${tokens.borderRadiusMedium})\`,
+
+buttonFontFamily: \`var(--buttonFontFamily,${tokens.fontFamilyBase})\`,
+
+...
+
+}
+// Button control tokens, used by Default button
+
+export const buttonTokens = {
+
+// Default
+
+ctrlButtonBorderRadius: \`var(--ctrlButtonBorderRadius, ${buttonGroupTokens.buttonBorderRadius})\`,
+
+ctrlButtonFontFamily: \`var(--ctrlButtonFontFamily,${buttonGroupTokens.buttonFontFamily})\`,
+
+...
+
+}
+
+```
+
+These would then be added to the current Theme object:
+
+```javascript
+// Tokens/src/types.ts
+import { ButtonTokens } from '@fluentui/react-components';
+
+export type Theme = FontSizeTokens &
+  LineHeightTokens &
+  BorderRadiusTokens &
+  StrokeWidthTokens &
+  HorizontalSpacingTokens &
+  VerticalSpacingTokens &
+  DurationTokens &
+  CurveTokens &
+  ShadowTokens &
+  ShadowBrandTokens &
+  FontFamilyTokens &
+  FontWeightTokens &
+  ColorPaletteTokens &
+  ColorStatusTokens &
+  ColorTokens &
+  ButtonTokens; // Addition
+```
+
```

Comment:
If I understand correctly, the suggestion is to store component-specific tokens inside the component package. For example, they would be placed in @fluentui/react-button. The @fluentui/tokens package would then import these tokens from @fluentui/react-components, while @fluentui/react-theme depends on @fluentui/tokens. This looks as kinda circular dependency chain 🤔 

Wouldn't it be better to store component-level tokens directly in the `@fluentui/tokens` package. In suggested approach any change will be a breaking behaviour. Plus, if in future there will be consideration for automation in tokens using tools like [style-dictionary](https://amzn.github.io/style-dictionary/#/), having all tokens in a single package could make easier their management